analyzers/tests/SonarAnalyzer.Test/TestCases/RouteTemplateShouldNotStartWithSlash.AspNet4x.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System.Web.Mvc;
+
+[Route("[controller]")]
+public class NoncompliantController : Controller // Noncompliant {{Change the paths of the actions of this controller to be relative and adapt the controller route accordingly.}}
+//           ^^^^^^^^^^^^^^^^^^^^^^
+{
+    [Route("/Index1")]                  // Secondary
+//   ^^^^^^^^^^^^^^^^
+    public ActionResult Index1() => View();
+
+    [Route("/SubPath/Index2_1")]        // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^
+    [Route("/[controller]/Index2_2")]   // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    public ActionResult Index2() => View();
+
+    [Route("/[action]")]                // Secondary
+//   ^^^^^^^^^^^^^^^^^^
+    public ActionResult Index3() => View();
+
+    [Route("/SubPath/Index4_1")]        // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^
+    [Route("/[controller]/Index4_2")]   // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    public ActionResult Index4() => View();
+}
+
+[RoutePrefix("[controller]")]
+public class NoncompliantWithRoutePrefixController : Controller // Noncompliant {{Change the paths of the actions of this controller to be relative and adapt the controller route accordingly.}}
+//           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+{
+    [Route("/Index1")]                  // Secondary
+//   ^^^^^^^^^^^^^^^^
+    public ActionResult Index1() => View();
+}
+
+[Route("[controller]")]
+[Route("[controller]/[action]")]
+public class NoncompliantMultiRouteController : Controller // Noncompliant {{Change the paths of the actions of this controller to be relative and adapt the controller route accordingly.}}
+//           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+{
+    [Route("/Index1")]                  // Secondary
+//   ^^^^^^^^^^^^^^^^
+    public ActionResult Index1() => View();
+
+    [Route("/SubPath/Index2_1")]        // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^
+    [Route("/[controller]/Index2_2")]   // Secondary
+//   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    public ActionResult Index2() => View();
+
+    [Route("/[action]")]                // Secondary
+//   ^^^^^^^^^^^^^^^^^^
+    public ActionResult Index3() => View();
+}
+
+[Route("[controller]")]
+public class CompliantController : Controller // Compliant: at least one action has at least a relative route
+{
+    [Route("/Index1")]
+    public ActionResult Index1() => View();
+
+    [Route("/SubPath/Index2")]
+    public ActionResult Index2() => View();
+
+    [Route("/[action]")]
+    public ActionResult Index3() => View();
+
+    [Route("/[controller]/Index4_1")]
+    [Route("SubPath/Index4_2")] // The relative route
+    public ActionResult Index4() => View();
```

Comment:
The `RouteAttribute` also has a parameterless constructor. Add a test case (a dedicated controller) for that as well.

Reply:
you mean this? https://github.com/SonarSource/rspec/blob/d9e454789cdfbdbb19c8266a106bfa32ed85f825/rules/S6931/S6931.AspNet4x.cs#L104

Reply:
In the end I added the rest of the parametrized UTs in this PR.
I realized it's confusing not to add all the tests at once for the reviewer.